lms/djangoapps/course_api/api.py

```diff
@@ -0,0 +1,58 @@
+"""
+Course API
+"""
+
+from django.contrib.auth.models import User
+from lms.djangoapps.courseware.courses import (
+    get_courses,
+    get_course_by_id
+)
+from opaque_keys.edx.keys import CourseKey
+
+from .serializers import CourseSerializer
+
+
+def _has_permission(requesting_user, username):
+    """
+    Returns true if `requesting_user` has permission to access the user
+    identified by `username`.
+
+    Arguments:
+        requesting_user (User): The user requesting permission to view another
+        username (string): The name of the user `requesting_user` would like
+        to access.
+
+    Return value:
+        Boolean
+    """
+    if not username:
+        return False
+    return requesting_user.is_staff or requesting_user.username == username
+
+
+def course_detail(course_key_string, request):
```

Comment:
I can see the logic in that.  But it does make for an inconsistent signature.  It requires callers to change the order of parameters around.  Also, as a reader, I usually ignore the initial `request` parameter as something that's part of the semantics of the method.  

But like I said, I'm not strongly opposed to reordering.